### ISSUE TITLE
Feature Request: Support XCFramework for All platforms in one

### ISSUE THREAD

**Issue by dreampiggy** [SDWebImage/SDWebImage#2762]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | All
 Platform Version        | All
 SDWebImage Version      | 5.1.0
 Integration Method      | carthage first
 Xcode Version           | Xcode 11

### Feature

Xcode 11 introduce the support for new framework structure, XCFramework. See : https://developer.apple.com/wwdc19/416

This `XCFramework` is a super combinataion of current `framework`. It can contains all the Apple platform's framework, including each Architecture (x86, x86_64, armv7, arm64, arm32_64, etc...), including each variant (UIKit for macOS and AppKit). And easy to use.

Now, for a multiple platform compatbiel framework, you don't need to add each `dependency` with 4 platforms or more. Just drag one instead.

For us, after #2743 was implements, the all framework combintation will bump to 6, make it little verbose to use. So this is a good solution for us.
